tests/rtdl/meson.build

```diff
@@ -0,0 +1,39 @@
+rtdl_test_cases = [
+	'dl_iterate_phdr',
+]
+
+foreach test_name : rtdl_test_cases 
+	test_rpath = meson.build_root() / 'tests' / 'rtdl' / test_name / ''
+	test_rpath += ':$ORIGIN/' # Workaround old and buggy qemu-user on CI
+
+	test_link_with = []
+	test_depends = []
+	test_native_link_with = []
+	test_native_depends = []
+
+	# Build the needed DSOs for the test. This sets the variables above.
+	subdir(test_name)
+
+	exec = executable('rtdl-' + test_name, [test_name / 'test.c', test_sources],
+		link_with: test_link_with + libc, include_directories: libc_include_dirs,
+		dependencies: rtlib_deps,
+		build_rpath: test_rpath,
+		override_options: test_override_options,
+		c_args: test_c_args,
+		link_args: test_link_args
+	)
+	test(test_name, exec, suite: 'rtdl', depends: test_depends)
+
+	if build_tests_host_libc and not host_libc_excluded_test_cases.contains(test_name)
+		exec = executable('host-libc-' + test_name, test_name / 'test.c',
+			link_with: test_native_link_with,
+			dependencies: rtlib_deps,
+			build_rpath: test_rpath,
+			override_options: 'b_sanitize=address,undefined',
+			c_args: ['-D_GNU_SOURCE', '-DUSE_HOST_LIBC', '-pthread'],
+			link_args: ['-lresolv', '-ldl', '-pthread', '-Wl,--disable-new-dtags'],
```

Comment:
rtdl tests don't need libresolv or pthread right? Also, what's up with `--disable-new-dtags` (and why isn't it needed when building the mlibc tests)?

Reply:
`--disable-new-dtags` forces the use of DT_RPATH vs the debian default of DT_RUNPATH. For some reason, dlopen doesn't seem to search DT_RUNPATH on the CI, so this flag makes it work (not really sure why).

Reply:
Alright, seems fine then.

Reply:
What about the extra deps?

Reply:
Yeah, I'll remove those, they were just copied from the other host tests.